### PR TITLE
JENKINS-45887 : Automatically archive JGiven reports

### PR DESCRIPTION
**Pull request by bguerin**

### CODE REVIEW

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    @Override
+    public void process(@Nonnull final StepContext context, @Nonnull final Element mavenSpyLogsElt)
+            throws IOException, InterruptedException {
+
+        TaskListener listener = context.get(TaskListener.class);
+        if (listener == null) {
+            LOGGER.warning("TaskListener is NULL, default to stderr");
+            listener = new StreamBuildListener((OutputStream) System.err);
+        }
+
```
Comment (cyrille-leclerc) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java]: Can we add a test to check the existence of JGiven reports before writing trying to load the Jenkins JGiven plugin and writing the message "[withMaven] Jenkins JGiven Plugin not found, do not archive jgiven reports."

Comment (bguerin) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java]: yep, of course !

Comment (bguerin) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java]: done

**Comment by bguerin** [comment]
@cyrille-leclerc changes done regarding your review. Ok for you now ?

**Comment by bguerin** [comment]
Hello @cyrille-leclerc 
Would you mind merging this ? Or tell me why not ?

**Comment by cyrille-leclerc** [comment]
@bguerin I felt guilty t merge before the CI build finished :-)
And there are failures in the build but it's another cause that I'mgoing to fix. The build takes so long on the https://ci.jenkins.io that I sometimes not check the results.

**Comment by cyrille-leclerc** [comment]
I made a mistake, I merged thinking that it was your toString() PR.

1. This PR breaks a unit test
2. Your code impacts the performances of every build doing a scan of the file system looking for jgiven files, even if the user doesn't use jgiven ([here}(https://github.com/jenkinsci/pipeline-maven-plugin/pull/71/files#diff-84979cb24f1c8b2768b965b06c6f8ff9R75). I would like to not impact builds that does not use JGiven.

@bguerin can you submit a new PR to fix the build and to not impact the performances of the builds that does not use JGiven?

**Comment by cyrille-leclerc** [comment]
thanks @bguerin !

For the performances, I thought that just checking the plugin existence before the file access could be a first step.

**Comment by bguerin** [comment]
Ok, it is what I was thinking about ...
But in your review, you said:

> Can we add a test to check the existence of JGiven reports before writing trying to load the Jenkins JGiven plugin and writing the message "[withMaven] Jenkins JGiven Plugin not found, do not archive jgiven reports.".
> We want to recommend to Jenkins users better usage of Jenkins to surface the valuable information of their Maven builds but we don't want to display recommendations for things that don't match with their usage of Jenkins & Maven.

and thus I add this file access ...

**Comment by bguerin** [comment]
Done, see #103

**Pull request merged**